lib/Serialization/SerializedModuleLoader.cpp

```diff
@@ -263,10 +263,43 @@ std::error_code SerializedModuleLoaderBase::openModuleDocFile(
   return std::error_code();
 }
 
+std::error_code
+SerializedModuleLoaderBase::openModuleSourceInfoFile(AccessPathElem ModuleID,
```

Comment:
This seems like duplicated code; can you factor out the common parts instead?

Reply:
hmm, it seems we have similar duplications for other two module inputs too. Do you have any ideas where we should move it to?

Reply:
I mean, here is fine, just generalize it.